src/PowerAuth/Context.cpp

```diff
@@ -216,6 +222,24 @@ ContextPtr Context::getInstance(ConfigurationPtr configuration)
     return context;
 }
 
+Context::Context(const Context& primary_context) :
+    _shared_mutex(primary_context.getSharedMutexPtr()),
+    _configuration(primary_context.getConfigurationPtr()),
+    _time_service(primary_context.getTimeServicePtr()),
+    _session_data(primary_context.getSessionDataPtr())
+{
+}
+
+ContextPtr Context::getTargetAlgorithmInstance(const ContextPtr& primary_context)
+{
+    auto config = primary_context->configuration();
+    auto spec = PowerAuthSpec::specForAlgorithm(config.algorithm());
+    
+    auto target_context = std::make_shared<Context>(*primary_context);
+    target_context->createServices(false, spec);
```

Comment:
This is a bit fishy for me. If created like this, then `sessionData` and `timeService` are not initialized in the target context. I wonder how it works, because `TimeService`  is definitely required in `AeadEncryptorFactory` and its encryptors.